meteor/lib/api/rundownLayouts.ts

```diff
@@ -29,7 +35,161 @@ export enum RundownLayoutsAPIMethods {
 	'createRundownLayout' = 'rundownLayout.createRundownLayout',
 }
 
+export interface LayoutDescriptor<T extends RundownLayoutBase> {
+	supportedElements: RundownLayoutElementType[]
+	filtersTitle?: string // e.g. tabs/panels
+	supportsFilters?: boolean
+}
+
+export interface CustomizableRegionSettingsManifest {
+	_id: string
+	title: string
+	layouts: Array<CustomizableRegionLayout>
+}
+
+export interface CustomizableRegionLayout {
+	_id: string
+	type: RundownLayoutType
+	filtersTitle?: string
+	supportsFilters?: boolean
+	supportedElements: RundownLayoutElementType[]
+}
+
+class RundownLayoutsRegistry {
```

Comment:
What's this? What is this used for?

Reply:
Right now it's just a datastore for the RundownLayoutsAPI to serve as a way to easily add new layout types, add support for a layout to a "region" (shelf/header etc), and register that a particular component is compatible with e.g. the Dashboard layout in the Shelf. My hope is that eventually we can "invert" the layouts system (certainly the dashboards) such that components are entriely self-describing (long long term). Therefore you could upload a bundle of components and have them register themselves as available in the Dashboard in the Shelf. Bit of a pipe dream but I think this helps to put us on that course.